src/movies.js

```diff
@@ -1,25 +1,175 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
 
+function getAllDirectors(moviesArray) {
+    let allDirectors=moviesArray.map(x=>x.director)
+   return allDirectors
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+  
+
+function howManyMovies(moviesArray) {
+  let result = 0
+  if (moviesArray.length===0){
+    return result
+  }
+
+  let moviesSpielberg = moviesArray.filter (movie => movie.director===`Steven Spielberg`)
+  let onlyDrama =[]
+  for ( let i = 0; i < moviesSpielberg.length; i ++){
+    let genre = moviesSpielberg[i].genre
+
+    for ( let j = 0; j < genre.length; j++ ) {
+      const element = genre[j];
+      
+      if (element==="Drama") {
+        result++
+        onlyDrama.push(moviesSpielberg[i]) 
+      }
+      
+    }
+    
+  }
+  
+
+ return result
+
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+function scoresAverage(moviesArray) {
+  let result = 0
```

Comment:
Esto lo puedes poner const ahora ya que no se reasigna (te quite la reasignacion! :D)
```suggestion
  const result = 0
```